tools/tarantool-gdb.py

```diff
@@ -397,14 +397,28 @@ def to_string_data(cls, data, depth):
 
         elif mp_type == cls.MP_STR:
             len = cls.decode_strl(data)
-            s += '"'
-            s += unicode(data.read(len), 'utf-8')
-            s += '"'
+            val = bytes(data.read(len))
+            # MP_STR may actually contain a binary string, in which case
+            # we encode the value in base64, like we do for MP_BIN.
+            is_bin = False
+            try:
+                val = unicode(val, 'utf-8')
+            except UnicodeDecodeError:
+                is_bin = True
+            if not is_bin:
+                s += '"'
+                # Escape backslash and double quotes.
+                s += val.replace('\\', '\\\\').replace('"', '\\"')
+                s += '"'
+            else:
+                s += '!!binary '
```

Comment:
```suggestion
                s += '!!binarystr '
```

Reply:
Added `bin:` and `str:` to binary strings and dropped `!!binary` tag, as discussed.